message/message.go

```diff
@@ -0,0 +1,61 @@
+package message
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/rs/zerolog"
+)
+
+// Source is the type used to represent messages source
+type Source string
+
+func (s Source) String() string {
+	return string(s)
+}
+
+// Type is the type used to represent messages type
+type Type string
+
+func (t Type) String() string {
+	return string(t)
+}
+
+// DataVersion is the type used to represent messages data version
+type DataVersion int
+
+// SchemaVersion is the type used to represent mesages schema version
+type SchemaVersion int
+
+const (
+	// SchemaVersion3 is the only schema version available right now.
+	// All messages are produced with whis as default.
+	SchemaVersion3 SchemaVersion = 3
+)
+
+// Message is the default event structure
+type Message struct {
+	SchemaVersion SchemaVersion
+	ID            string `json:"Id"`
+	Source        Source
+	Type          Type
+	CreatedAt     string
+	DataVersion   DataVersion
+	Data          json.RawMessage
+
+	// A context should be given to the message when it is created. If this context
+	// is overwritten (i.e context.WithValue), be sure of possible racing
+	// conditions.
+	Context context.Context `json:"-"`
```

Comment:
I think that this Context was used mainly to transport pubsub acking information through the system. If someone using FoundationKit's message would need this context, it is possible to import the Message structure as a mixin. Thus, I believe that Context should not be here.